sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -725,16 +725,17 @@ def rollback(self) -> None:
         """Rollback to the previous migration."""
         logger.info("Starting migration rollback.")
         tables = (self.snapshots_table, self.environments_table, self.versions_table)
-        if not any(self.engine_adapter.table_exists(f"{table}_backup") for table in tables):
-            raise SQLMeshError("There are no prior migrations to roll back to.")
-        for table in tables:
-            self._restore_table(table, _backup_table_name(table))
-
-        if self.engine_adapter.table_exists(_backup_table_name(self.seeds_table)):
-            self._restore_table(self.seeds_table, _backup_table_name(self.seeds_table))
-
-        if self.engine_adapter.table_exists(_backup_table_name(self.intervals_table)):
-            self._restore_table(self.intervals_table, _backup_table_name(self.intervals_table))
+        versions = self.get_versions(validate=False)
+        if versions.schema_version == 0:
+            # Clean up state tables
+            for table in tables + (self.seeds_table, self.intervals_table):
+                self.engine_adapter.drop_table(table)
+        else:
+            if not any(self.engine_adapter.table_exists(f"{table}_backup") for table in tables):
+                raise SQLMeshError("There are no prior migrations to roll back to.")
+            for table in tables + (self.seeds_table, self.intervals_table):
+                if self.engine_adapter.table_exists(_backup_table_name(table)):
```

Comment:
The reason why `intervals` and `seeds` tables were handled separately is because they were added in later versions so we didn't expect backups to exist for them. Otherwise, checking table existence twice seems redundant.

Reply:
Got it. I'll revert this then.

Reply:
@izeigerman I changed line 734 from `any()` to `all()`. If some but not all the backups are there, then it's in a weird state and we shouldn't drop any tables.